src/makeInfinityAdder.js

```diff
@@ -3,8 +3,24 @@
 /***
  * @return {function}
  */
-function makeInfinityAdder() {
-  // write code here
+function makeInfinityAdder(a) {
+  let sum = 0;
+
+  function inner(b) {
+    if (b === undefined) {
```

Comment:
The check for `undefined` is unnecessary as per the task's additional rules. The `inner` function should only check if no arguments are passed to decide when to return the sum and reset it.